### PR TITLE
tests: add integ test for `sagemaker-studio` module

### LINKED ISSUE #160: [CHORE] Module modules/sagemaker/sagemaker-studio missing integration tests

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
## Describe your changes
- add integ test for `sagemaker-studio` module

## Issue ticket number and link
- closes #160

## Checklist before requesting a review

- [x] I updated `CHANGELOG.MD` with a description of my changes
- [x] If the change was to a module, I ran the code validation script (`scripts/validate.sh`)
- [x] If the change was to a module, I have added thorough tests
- [x] If the change was to a module, I have added/updated the module's README.md
- [x] If a module was added, I added a reference to the module to the repository's README.md
- [x] I verified that my code deploys successfully using `seedfarmer apply`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 3cc8bba9a784f4a121c2b886458ea87f32e803d1
* Result: FAILED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=62f0c737-143a-4700-9188-da5bf14e32ff%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 979b7bb44af97828aa4c4603935c1e4d76a9bfaf
* Result: FAILED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ccd702e3-1795-4d5b-a335-639bcdeb049f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: eeff9b22f91376ff8fe1b326726f185d783ccb7b
* Result: FAILED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=eb30faa0-4ef3-467d-aa39-1ffe28828905%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 52a75a79339138b484f3c9a281a5faabaaaf1fd2
* Result: FAILED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=d806c443-eadf-424d-9545-3af0bde91db6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 2d78980d3f9a25360ab649d17911dbfe2d1e0475
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=8eee4dfe-981b-4954-9692-a1ec51ea54fd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 11354874f11d4a05ed8edcdc716fb127aad4429b
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=f47cb2fa-9946-4b6b-9c66-506c1a85ee32%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/sagemaker/sagemaker-studio/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
-aws-cdk-lib==2.128.0
-aws_cdk.aws_lambda_python_alpha==2.128.0a0
-cdk-nag==2.28.40
-pydantic~=2.7.1
-pydantic-settings~=2.2.1
+aws-cdk-lib~=2.149.0
```
Comment (kukushking) [inline modules/sagemaker/sagemaker-studio/requirements.txt:1]: Looks like we need to update deployspec with the new CDK version too

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: ff7763540939665bf00cce0c32334416c213ab2e
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=eb98e176-26e4-4a98-b2bd-8aed378fd5ed%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]
Thank you!

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 878a2d5ef7c090463ac82cd45f1d7b1e846d048f
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=72c1893b-2b15-4de1-8472-ce0ffc50ea93%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: c096701e0530efba4a082290f54f1cf09b780880
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=cf8cff6b-29cd-4fa7-ab8a-6c352d17ff90%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**